### DISCUSSION TITLE
Failed to locate 'git', requesting installation of command line developer tools. Failed during: git init -q

### DISCUSSION THREAD

**Discussion by Fer-CV** [Homebrew/discussions#3694] [Getting started]
Hi

I am not used with these command things, so I hope someone can help me!
When I want to install homebrew (via brew.sh) I copied the command, enter and entered my password than I get:

```
 sh: line 1: 51260 Bus error: 10           /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null
git: error: sh -c '/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null' failed with exit code 35328: (null) (errno=Invalid argument)
xcode-select: Failed to locate 'git', requesting installation of command line developer tools.
Failed during: git init -q
```

So what to do? ( I need it to publish pageflip PDF with the aid of Ghostscript, so I get neat clean text)

Thanks!

**Comment by SMillerDev** [comment]
It must have given a popup asking for confirmation to install the command line developer tools. Accept the install and then it can continue.

**Reply by Fer-CV** [reply]
That I did, but I get the usual start in the terminal

**Reply by SMillerDev** [reply]
What happens when you run `git status`?

**Comment by Fer-CV** [comment]
```console
sh: line 1: 52797 Bus error: 10           /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null
git: error: sh -c '/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null' failed with exit code 35328: (null) (errno=Invalid argument)
xcode-select: Failed to locate 'git', requesting installation of command line developer tools.
```

**Comment by Fer-CV** [comment]
Hi Sean

Thanks for your help!

 
I get:

```
~ ferdinand$ git status
sh: line 1: 52797 Bus error: 10           /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null

git: error: sh -c '/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null' failed with exit code 35328: (null) (errno=Invalid argument)

xcode-select: Failed to locate 'git', requesting installation of command line developer tools.
```

Best!

Ferdinand

**Reply by SMillerDev** [reply]
It seems your shell runs everything through `xcodebuild`, is your device managed by a company or educational institution? Because this is not the default behaviour of these commands.

**Reply by Fer-CV** [reply]
Hi 
Thanks, but I use my own iMac 27" 2017, 40GB

**Comment by alanhoyle** [comment]
I bet if you scroll up, in your brew you'll find something like this:

```
Please ensure Xcode packages are up-to-date — try running 'xcodebuild -runFirstLaunch'.
```

Try running that.  It fixed a similar issue for me.

**Reply by Fer-CV** [reply]
Hi
Thanks, I didn't find it, but I ran the command, without any result when running the install
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
I still get
sh: line 1: 80763 Bus error: 10           /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null
git: error: sh -c '/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -find git 2> /dev/null' failed with exit code 35328: (null) (errno=Invalid argument)
xcode-select: Failed to locate 'git', requesting installation of command line developer tools.
Failed during: git init -q

**Reply by SMillerDev** [reply]
What macOS version are you using?

**Reply by Fer-CV** [reply]
I ran the xcodebuild -runFirstLaunch command, and than the homebrew install command but no result.
I installed xcode commands etc > no result
I restarted my computer > no result

I ran 12.6 Montery.........

**Reply by Praktiker** [reply]
Hi, got the same problem.
Solution was to start Xcode on my mac studio and filled out Developer Name as Xcode instructed me after I opened a test app.
After that the home-brew installation routine worked like a charm.
regards Praktiker.

**Reply by RaphaelBajon** [reply]
Hi, 
I had the same issues. Just by doing the update asking on Xcode it goes right for myself. I hope that's fine for you.

**Reply by arvinxx** [reply]
work for me. Thanks!
